### PR TITLE
Use MAC from EEPROM for Interfaces

### PR DESCRIPTION
**Pull request by andriymoroz-mlnx**
Signed-off-by: Andriy Moroz <c_andriym@mellanox.com>

**- What I did**
Updated syncd init script to pass MAC from EEPROM to SAI for Mellanox platforms. Interfaces will be created with this MAC

**- How I did it**
updated syncd init script (syncd_init_common.sh) 

**- How to verify it**
Start SONiC and make sure MACs on Interfaces are similar to one returned by command
od -vt x1 -An /sys/bus/i2c/devices/8-0051/eeprom | xargs printf "0x%s " | xargs printf "%02x:" | awk 'BEGIN { FS=":"; i=8+1+2+1} {while(i<NF) {type=$i; len=("0x"$(i+1));if(type!="24") {i=i+2+len} else {print substr($0, (i+1)*3+1, len*3-1); break}}}'

some bits in last byte may differ

### CODE REVIEW

**Comment by qiluo-msft** [comment]
Please fix the test.

File: syncd/scripts/syncd_init_common.sh [hunk 1]
```diff
@@ -54,7 +54,7 @@ config_syncd_mlnx()
     [ -e /dev/sxdevs/sxcdev ] || ( mkdir -p /dev/sxdevs && mknod /dev/sxdevs/sxcdev c 231 193 )
 
     # Read MAC address and align the last 6 bits.
-    MAC_ADDRESS=$(ip link show eth0 | grep ether | awk '{print $2}')
+    MAC_ADDRESS=$(od -vt x1 -An /sys/bus/i2c/devices/8-0051/eeprom | xargs printf "0x%s " | xargs printf "%02x:" | awk 'BEGIN { FS=":"; i=8+1+2+1} {while(i<NF) {type=$i; len=("0x"$(i+1));if(type!="24") {i=i+2+len} else {print substr($0, (i+1)*3+1, len*3-1); break}}}')
```
Comment (qiluo-msft) [inline syncd/scripts/syncd_init_common.sh]: Consider to use python -c to simplify the manipulation.

**Comment by andriymoroz-mlnx** [comment]
ok
looks like it may take some time...
as a workaround to resolve the issue you can manually fix syncd:/usr/bin/syncd_init_common.sh on the switch and restart it

**Comment by andriymoroz-mlnx** [comment]
retest this please

File: syncd/scripts/syncd_init_common.sh [hunk 2]
```diff
@@ -54,7 +54,7 @@ config_syncd_mlnx()
     [ -e /dev/sxdevs/sxcdev ] || ( mkdir -p /dev/sxdevs && mknod /dev/sxdevs/sxcdev c 231 193 )
 
     # Read MAC address and align the last 6 bits.
-    MAC_ADDRESS=$(ip link show eth0 | grep ether | awk '{print $2}')
+    MAC_ADDRESS=$(sonic-cfggen -d -v DEVICE_METADATA.localhost.mac)
     last_byte=$(python -c "print '$MAC_ADDRESS'[-2:]")
```
Comment (andriymoroz-mlnx) [inline syncd/scripts/syncd_init_common.sh]: probably I can remove last byte masking as it is already done for the MAC in DB

**Pull request merged**